pkg/controller/testing/table.go

```diff
@@ -265,6 +268,13 @@ type Ctor func(*Listers, controller.Options) controller.Interface
 func (r *TableRow) Test(t *testing.T, ctor Ctor) {
 	ls := NewListers(r.Objects)
 
+	for i, action := range r.WantUpdates {
+		if builder, ok := action.Object.(builder.ConfigBuilder); ok {
```

Comment:
Seems like this could avoid the reference to `ConfigBuilder` by defining an interface:

```go
type Builder interface{
  Build() runtime.Object
}
```

but I'm not sure about that return value. Would it work with `cmp.Diff` or semantic equality or whatever is being used to compare objects?